### PR TITLE
Add Terraform Example

### PR DESCRIPTION
**Pull request by dchw**

### CODE REVIEW

File: examples/terraform/start_localstack.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+docker run -d -p 4566:4566 -e SERVICES=s3 --name localstack --network="host" localstack/localstack:0.11.5
```
Comment (vladaionescu) [inline examples/terraform/start_localstack.sh:3]: We could reuse this file within the Earthfile if we wanted to. Or we could remove it from this example perhaps.

File: examples/terraform/Earthfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
+build:
```
Comment (vladaionescu) [inline examples/terraform/Earthfile:2]: We should add this example to the root Earthfile, under the `+examples` target. That'll cause the CI to pick it up.

**Review by vladaionescu — COMMENTED** [review]
Looks good! Left two minor comments.

**Review by agbell — APPROVED** [review]
Great example!

File: examples/terraform/Earthfile (hunk 2 above)
Comment (dchw) [reply examples/terraform/Earthfile:2]: Yep, realized I spaced that. Fixed.

File: examples/terraform/Earthfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    FROM hashicorp/terraform:light
+
+    ARG AWS_ACCESS_KEY_ID
+    ARG AWS_SECRET_ACCESS_KEY
```
Comment (vladaionescu) [inline examples/terraform/Earthfile:6]: We should probably add a comment here saying that in prod environments we should use a secret. However in this case this is not really a secret so it's ok.

**Review by vladaionescu — APPROVED** [review]
:+1:

File: examples/terraform/Earthfile (hunk 3 above)
Comment (dchw) [reply examples/terraform/Earthfile:6]: Yeah, I was also looking at some of the terraform and some comments are probably in order.

**Pull request merged**